src/handlers/hunk_header.rs

```diff
@@ -249,11 +249,15 @@ fn parse_hunk_header(line: &str) -> Option<ParsedHunkHeader> {
                 )
             })
             .collect();
-        let code_fragment = caps[2].to_string();
-        Some(ParsedHunkHeader {
-            code_fragment,
-            line_numbers_and_hunk_lengths,
-        })
+        if line_numbers_and_hunk_lengths.len() == 0 {
+            None
+        } else {
+            let code_fragment = caps[2].to_string();
+            Some(ParsedHunkHeader {
+                code_fragment,
+                line_numbers_and_hunk_lengths,
+            })
+        }
```

Comment:
I'm a little unsure if this is the most idiomatic Rust!

Reply:
Well clippy got me to use `is_empty()`, but still I'm unsure on this `if` structure with all these implicit returns...

Reply:
This is fine, `caps[2]` was the original non-idiomatic line. Now at least it does not crash. Thanks!